File: python/Veo/VeoGenerateGeminiAI.py
import time
from google import genai
from google.genai.types import GenerateVideosConfig, Image

client = genai.Client(api_key="xxx")

def veo_download_local():
    operation = client.models.generate_videos(
        model="veo-2.0-generate-001",
        prompt="a cat reading a book",
        config=GenerateVideosConfig(
            aspect_ratio="16:9",
            number_of_videos=1,
        ),
    )

    while not operation.done:
        time.sleep(15)
        operation = client.operations.get(operation)
        print(operation)

    # Download the video.
    video = operation.response.generated_videos[0]
    client.files.donwload(file=video.video)
    video.video.save("veo3.1_with_reference_images.mp4")
    print("Generated video saved to veo3.1_with_reference_images.mp4")

File: java/src/main/java/Veo/VeoGenerateGeminiAI.java
package Veo;

import com.google.genai.Client;
import com.google.genai.types.GenerateVideosConfig;
import com.google.genai.types.GenerateVideosOperation;

public class VeoGenerateGeminiAI {

    public static void main(String[] args) {

        Client client = Client.builder()
                .apiKey("xxx")
                .build();

        veoDownloadLocal(client);

    }

    public static void veoDownloadLocal(Client client){

        GenerateVideosConfig config = GenerateVideosConfig.builder()
                .aspectRatio("16:9")
                .numberOfVideos(1)
                .build();

        GenerateVideosOperation operation = client.models.generateVideos(
                "veo-2.0-generate-001",
                "a cat reading a book",
                null,
                config
        );

        while (!operation.done().isPresent()) {
            try {
                System.out.println("Waiting for operation to complete...");
                Thread.sleep(10000);
                // Sleep for 10 seconds and check the operation again
                operation = client.operations.getVideosOperation(operation, null);
            } catch (InterruptedException e) {
                System.out.println("Thread was interrupted while sleeping.");
                Thread.currentThread().interrupt();
            }
        }

        operation.response().ifPresent(
                response -> {
                    response.generatedVideos().ifPresent(
                            videos -> {
                                System.out.println("Generated " + videos.size() + " videos.");
                                videos.get(0).video().ifPresent(video -> {
                                    // 调用下载方法，将视频保存为本地文件
                                    client.files.download(video,"neon_cat_video.mp4",null);
                                });
                            }
                    );
                }
        );
    }

}
